pandas/io/common.py

```diff
@@ -712,7 +713,17 @@ def get_handle(
 
     # Convert BytesIO or file objects passed with an encoding
     is_wrapped = False
-    if is_text and (compression or _is_binary_mode(handle, ioargs.mode)):
+    if not is_text and ioargs.mode == "rb" and isinstance(handle, TextIOBase):
+        handle = BytesIOWrapper(
+            handle,
+            encoding=ioargs.encoding,
+        )
+        handles.append(handle)
+        # Handle is always provided by caller since
+        # they are trying to read bytes from string
+        # buffer
+        is_wrapped = True
```

Comment:
Maybe change the comment to something like "the (text) handle is always provided by the caller since get_handle would have opened it in binary mode". I am not sure whether that is clearer.